docs/features/mcl.md

```diff
@@ -0,0 +1,85 @@
+# Setting up multi cluster support for KCC
+
+> NOTE: This feature is in alpha stage and is actively being worked on.
+
+Imagine we have two Kubernetes clusters, `mcl-cluster-1` and `mcl-cluster-2`. We want to run KCC in both for high availability, but only one instance should be the active leader at any time.
+
+### Setup
+
+#### 1. Cluster Lock: Powered GCS Bucket
+
+```sh
+# This is a one-time setup
+gcloud storage buckets create gs://kcc-global-leader-lock
+```
+
+#### 2. The `multiclusterlease-controller` (MCL Controller)
+
+Deploy the MCL controller to **both** `mcl-cluster-1` and `mcl-cluster-2`. This controller's job is to watch for `MultiClusterLease` CRs and contend for the lock in the GCS bucket.
+
+```yaml
+# Deployed in BOTH mcl-cluster-1 and mcl-cluster-2
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: multiclusterlease-controller-manager
+  namespace: multiclusterlease-system
+spec:
+  replicas: 1
+  template:
+    spec:
+      # This controller needs GCP credentials to talk to the bucket.
+      # This is best done using Workload Identity, which binds the Kubernetes
+      # Service Account (KSA) used by this pod to a Google Service Account (GSA).
+      # Ensure the KSA (e.g., 'default' in 'multiclusterlease-system') is
+      # configured for Workload Identity.
+      serviceAccountName: default
+      containers:
+      - name: manager
+        image: gcr.io/multiclusterlease/controller:latest
+        command:
+        - /manager
+        args:
+        - --gcs-bucket=kcc-global-leader-lock
+```
+
+At this point, we have an election controller running in each cluster, ready to act on behalf of any client that creates a `MultiClusterLease` resource.
```

Comment:
Where does the 'election controller' come in to this. I couldn't find a definition of it any earlier in the doc. Seems like you mean the 'multiclusterlease-controller' but why change the name?